### PR TITLE
[REFACTOR] Menu for flexibility, consistency and one-liner code

### PR DESCRIPTION
**Pull request by georgehristov**

### CODE REVIEW

File: src/Menu.php [hunk 1]
```diff
@@ -93,45 +91,44 @@ public function addHeader($name)
...
-            unset($name[0]);
-        } else {
-            $label = $name;
-            $name = [];
-        }
+        $subMenu = $this->add([new self(), 'defaultTemplate' => 'submenu.html', 'ui' => 'dropdown', 'in_dropdown' => true]);
 
-        $sub_menu = $this->add([new self(), 'defaultTemplate' => 'submenu.html', 'ui' => 'dropdown', 'in_dropdown' => true]);
-        $sub_menu->set('label', $label);
+        $name = (array) $name;
```
Comment (DarkSide666) [inline src/Menu.php]: again, what if `$name===null` ?

File: src/Menu.php [hunk 2]
```diff
@@ -93,45 +91,44 @@ public function addHeader($name)
...
+
+        if ($label = $name['title'] ?? $name['label'] ?? $name['name'] ?? $name[0] ?? '') {
```
Comment (DarkSide666) [inline src/Menu.php]: what if `$label === 0` or `$label === '0'`?

File: src/Menu.php [hunk 3]
```diff
@@ -93,45 +91,44 @@ public function addHeader($name)
...
      *
      * @return Menu
      */
-    public function addGroup($title)
+    public function addGroup($name)
     {
         $group = $this->add([new self(), 'defaultTemplate' => 'menugroup.html', 'ui' => false]);
-        if (is_string($title)) {
+
+        $name = (array) $name;
```
Comment (DarkSide666) [inline src/Menu.php]: same issue

File: src/Menu.php [hunk 4]
```diff
@@ -93,45 +91,44 @@ public function addHeader($name)
...
+
+        if ($title = $name['title'] ?? $name['label'] ?? $name['name'] ?? $name[0] ?? '') {
```
Comment (DarkSide666) [inline src/Menu.php]: and theoretically $title can be zero or '0'

File: src/Menu.php [hunk 5]
```diff
@@ -36,19 +36,17 @@ class Menu extends View
...
      *
      * @return Item
      */
     public function addItem($item = null, $action = null)
     {
-        if (is_string($item)) {
-            $item = ['Item', $item];
-        } elseif (is_array($item)) {
+        if (!is_object($item)) {
+            $item = (array) $item;
```
[the remark this answers is not in the document]
Comment (georgehristov) [reply src/Menu.php]: `(array) null` converts to empty array

File: src/Menu.php (hunk 1 above)
Comment (georgehristov) [reply src/Menu.php]: converts to empty array

File: src/Menu.php (hunk 3 above)
Comment (georgehristov) [reply src/Menu.php]: converts to empty array

File: src/Menu.php (hunk 4 above)
Comment (georgehristov) [reply src/Menu.php]: fixed with a6b535b

File: src/Menu.php (hunk 2 above)
Comment (georgehristov) [reply src/Menu.php]: fixed with a6b535b

File: src/Menu.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (DarkSide666) [reply src/Menu.php]: Ouch, my mistake :(  You're right.

**Pull request merged**